Argo docs update (#350)
nebari-dev/nebari-docs: docs/docs/how-tos/using-argo.md

---
id: using-argo
title: Automate your first workflow with Argo
description: Argo workflow management
---

# Automate workflows with Argo

[Argo Workflows](https://argoproj.github.io/workflows) is an open source container-native
workflow engine for orchestrating parallel jobs on Kubernetes. In other words,

> Argo helps you run a sequence on tasks or functions without you having to be

Reviewer: ```suggestion
Argo helps you run a sequence of tasks or functions without you having to be
```

present (it will manage the server resources for you). Argo workflows
comes enabled by default with Nebari deployments.

Access control for Argo on Nebari is done through Keycloak user groups. All

> users in the `admin` or `developer` groups have access to Argo.

Reviewer: ```suggestion
users in the `admin` or `developer` groups have access to Argo.

:::note
Also see the [Set up Argo Workflows documentation](/how-tos/setup-argo).
:::

```

> ## Ways to access the Argo Server

Reviewer: ```suggestion
## Access the Argo Server
```

the landing page. Argo offers a template for the workflow yaml format.

![Argo UI submit new workflow](/img/how-tos/argo_submit_new_workflow.png)

Click `+ CREATE` when you're ready to submit. The yaml format is not the only
option for generating workflows. Argo also allows you to create workflows via
python. More information on how to generate these specifications will follow.

## Submit a workflow via Argo CLI

You can also submit or manage workflows via the Argo CLI. The Argo CLI can be
downloaded from the
[Argo Releases](https://github.com/argoproj/argo-workflows/releases) page.

You can submit a workflow through the CLI using `argo submit my-workflow.yaml`.

The `argo list` command will list all the running workflows.

If you've just submitted a workflow and you want to check on it, you can run
`argo get @latest` to get the latest submitted workflow.

You can also access the logs for a workflow using
`argo logs -n workflow_name @latest`.

For more information on Argo workflows via the UI or the CLI, you can visit the
[Argo docs](https://argoproj.github.io/argo-workflows/workflow-concepts/).

[Hera](https://hera-workflows.readthedocs.io/) is a framework for building and

> submitting Argo workflows in Python. TODO - reference another doc.

Reviewer: ```suggestion
submitting Argo workflows in Python. Learn more in the [Argo Workflows walkthrough tutorial](/tutorials/argo-workflows-walkthrough).
```

## Access your Nebari environments and file system while on an Argo pod (BETA)

Once you move beyond the "Hello World" Argo examples, you may realize that the
conda environments and the persistent storage you have on Nebari would be
really useful in your temporary Argo pods. Lucky for you, we've solved that
problem for you!

> Nebari comes with [Nebari Workflow Controller (BETA)](https://github.com/nebari-dev/nebari-workflow-controller)(NWC)

Reviewer: What does NWC mean?

> Nebari comes with [Nebari Workflow Controller (BETA)](https://github.com/nebari-dev/nebari-workflow-controller)(NWC)

Reviewer: ```suggestion
Nebari comes with [Nebari Workflow Controller (BETA)](https://github.com/nebari-dev/nebari-workflow-controller), abbreviated as NWC,
```

which transfers the user's environment variables, home and shared directories,
docker image, and available conda environments to the server where the Workflow
is running. Users can then run a script that loads and saves from their home
directory with a particular conda environment.

All of these things are enabled when users add the `jupyterflow-override` label
to their workflow as in this example using Hera:

> ```

Reviewer: ```suggestion
```python
```

---
id: using-argo
title: Automate your first workflow with Argo
description: Argo workflow management
---

# Automate workflows with Argo

[Argo Workflows](https://argoproj.github.io/workflows) is an open source container-native
workflow engine for orchestrating parallel jobs on Kubernetes. In other words,
Argo helps you run a sequence of tasks or functions without you having to be
present (it will manage the server resources for you). Argo workflows
comes enabled by default with Nebari deployments.

Access control for Argo on Nebari is done through Keycloak user groups. All
users in the `admin` or `developer` groups have access to Argo.

:::note

> Also see the [Set up Argo Workflows documentation](/how-tos/setup-argo).

Reviewer: ```suggestion
Also see the [Set up Argo Workflows documentation](/docs/how-tos/setup-argo).
```

the landing page. Argo offers a template for the workflow yaml format.

![Argo UI submit new workflow](/img/how-tos/argo_submit_new_workflow.png)

Click `+ CREATE` when you're ready to submit. The yaml format is not the only
option for generating workflows. Argo also allows you to create workflows via
python. More information on how to generate these specifications will follow.

## Submit a workflow via Argo CLI

You can also submit or manage workflows via the Argo CLI. The Argo CLI can be
downloaded from the
[Argo Releases](https://github.com/argoproj/argo-workflows/releases) page.

You can submit a workflow through the CLI using `argo submit my-workflow.yaml`.

The `argo list` command will list all the running workflows.

If you've just submitted a workflow and you want to check on it, you can run
`argo get @latest` to get the latest submitted workflow.

You can also access the logs for a workflow using
`argo logs -n workflow_name @latest`.

For more information on Argo workflows via the UI or the CLI, you can visit the
[Argo docs](https://argoproj.github.io/argo-workflows/workflow-concepts/).

[Hera](https://hera-workflows.readthedocs.io/) is a framework for building and

> submitting Argo workflows in Python. Learn more in the [Argo Workflows walkthrough tutorial](/tutorials/argo-workflows-walkthrough).

Reviewer: ```suggestion
submitting Argo workflows in Python. Learn more in the [Argo Workflows walkthrough tutorial](/docs/tutorials/argo-workflows-walkthrough).
```
